src/react-apps/applications/runtime/src/utils/validation.ts

```diff
@@ -43,13 +44,45 @@ const validationFunctions: any = {
   pattern,
 };
 
+export function validateFormComponents(
+  formAttachments: any,
+  formLayout: any,
+  language: any,
+) {
+  const validations = {};
+  const fieldKey = 'simpleBinding';
+  const attachments = formAttachments ? Object.keys(formAttachments).length : 0;
+  formLayout.forEach((component) => {
+    validations[component.id] = {};
```

Comment:
Final question, is it necessary to initialize the key it if no validation messages are added? I figure this will trigger an unnecessary re-render (?). Should probably only initialize key if it actually are any validation messages for a given key.

Reply:
Good point 👍